### DISCUSSION TITLE
android aab compiling with 33.1.1.533 sdk missing runtime.apk

### DISCUSSION THREAD

**Discussion by mehmury** [airsdk/Adobe-Runtime-Support#938] [Q&A]
I tried the latest air sdk for android aab packaging. I got stuck at one last stage.
I applied the commands from cmd with adt and I am getting the error of missing runtime.

SDK is missing file air/android/device/x86/Runtime.apk

The runtime.apk file appears to exist at the location specified in the air sdk.

What could be the cause of this error?

**Comment by ajwfrost** [comment]
Can you please first check the file size - should be 14,099,661 bytes; the other thing that could cause that output is if the AAPT tool fails. So can you please try running:
```
[SDK folder]\lib\android\bin\aapt.exe l -a [SDK folder]\runtimes\air\android\device\x86\Runtime.apk
```
and check that this works properly?

**Reply by mehmury** [reply]
thanks for reply.i try this process on cmd and try again adt compiling ..
but i got same error ..
i use myflashlabs firebase and admob anes.
i wonder needs update new version for anes?
i use now last versions that anes.
14,099,661 bytes mean 15 mb.?
this not enoug for me...

**Reply by ajwfrost** [reply]
Did that AAPT command output a large amount of text when you ran it?

And yes, the .apk file should be around 15MB - actually no, sorry, 13.4MB, see below..: if yours is a different size then something may have gone wrong with the SDK copy, so please try to check the downloaded .zip file again and see if you can extract that file.
![image](https://user-images.githubusercontent.com/6400804/124124134-fdd8c900-da6f-11eb-802f-bf7e89d2832a.png)

13,770 KB = 14,100,480 bytes = 13.44MB. I must have been looking at the wrong file to say it was 15MB, sorry!

**Reply by mehmury** [reply]
oh,sorry !! 
I thought it was my app size. The correct runtime.apk file size is 13,770 mb. No problem here. I guess it's not the file size that caused me to get this error.
Could it be that myflashlabs ane doesn't support x86 output?

**Reply by ajwfrost** [reply]
If it's the ANE not including support for x86, then that's not the error you should see. The "SDK is missing file xxx" is a particular type of exception, if there are problems processing ANEs then you'd get an "invalid input" exception. Besides which I think missing ANE platforms aren't a problem, the build should proceed and at runtime AIR will just pick out the "default" SWF from the ANE if it can't find the right platform-specific version...

We maybe will have to add some extra debugging in here. My only short-term suggestion would be to re-download the SDK and copy it somewhere else, in case there's a file corruption problem.

Also, if you can send us the output of that AAPT command, we can check that because we do parse it for a version string, so if that's missing/malformed it may cause the same sort of problem..

thanks

**Comment by mehmury** [comment]
Thanks.i try download zip file again...

i attached a text file for AAPT check from cmd on my device ..i will follow the results.

 
[appt_check_1july2021.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/6748375/appt_check_1july2021.txt)

**Reply by ajwfrost** [reply]
Okay - so actually that looks like it's fine, for that to output the details like that and to correctly show the version string, the apk file must be okay!
`A: android:versionName(0x0101021c)="33.1.1.533" (Raw: "33.1.1.533")`

We'll get a version out with some more debugging to help work this out....

**Reply by mehmury** [reply]
okay.thanks..

**Reply by ajwfrost** [reply]
So: can you please download the updated ADT.jar file from the below link and put this into the SDK's "lib" folder.
You'll also find the "adt.cfg" file in here: you need to remove the comment at the start of the "DebugOut=true" line so that you get a debug file being created...
https://transfer.harman.com/message/rGFdqpUg3kHplopv095H4o

Then once you run it again, you should find a file "adt.log" in your user/home folder (e.g. "c:\users\username" or "~" on a mac) - please let us know what it says!

thanks

**Reply by mehmury** [reply]
Thanks Andrew. I downloaded the adt.jar file from the link you gave, added it to the sdk and got the debug records. I am adding the text file.
Waiting for your return.
[adt.log](https://github.com/airsdk/Adobe-Runtime-Support/files/6749574/adt.log)

**Reply by ajwfrost** [reply]
Okay thanks... looking at it, the only thing I can think of is that it can't actually resolve the SDK's folder properly! Might need to add in some extra debugging here...

**Reply by mehmury** [reply]
I tried a few more tests with the sdk.
When I turn on unicode support in the unicode section of windows 10 language settings, it does not recognize the android sdk path we give as platformsdk.
Could the error I get be indirectly related to the language settings?

**Reply by ajwfrost** [reply]
Yes it may be! Can you try with the below ADT - worth clearing the log file first and then run it once, there's a lot of output in there I suspect...
https://transfer.harman.com/message/aYlZTyPX6FJ3D1xqiLlORC

thanks

**Reply by mehmury** [reply]
I tried again, posting the adt.log file.
Am I the only one having this problem?
I hope it is not an insoluble situation. I want to be ready for the google aab transition in August.

waiting for your feedback 
[adt.log](https://github.com/airsdk/Adobe-Runtime-Support/files/6749865/adt.log)

**Reply by ajwfrost** [reply]
Thanks... so with a bit of formatting:
```
Got runtime path = C:\AIRSDK_3311533REAL\runtimes\air\android\device\x86\Runtime.apk
Created zip file= java.util.zip.ZipFile@cbafc2
Exception: java.io.IOException: Unable to find source.properties
```

So .. found it. This is happening because you have a folder within your Android SDK's `build-tools` folder, which then doesn't have a `source.properties` file. We should ignore such folders I guess!

Worth checking that you do actually have a valid build tools folder e.g. `30.0.2` that contains a `source.properties` file? If you clear out everything else, then it should work....

thanks

**Reply by mehmury** [reply]
you are great!!! Thanks a lot Andrew !!! problem solved. but i got only a warning about java version.which version of java need for JAVA_HOME path this air sdk?

NOTE: Multiple SDKs exist, choosing version 31.0.0 rc2 for packaging
NOTE: Multiple SDKs exist, choosing version 31.0.0 rc2 for packaging
NOTE: Multiple SDKs exist, choosing version 31.0.0 rc2 for packaging
NOTE: Multiple SDKs exist, choosing version 31.0.0 rc2 for packaging
NOTE: Multiple SDKs exist, choosing version 31.0.0 rc2 for packaging
JAVA_HOME does not point to a valid JDK installation, which is required for Android App Bundle creation

**Reply by ajwfrost** [reply]
Excellent - we'll change ADT so that it can cope with this scenario...

For JAVA_HOME I think anything 1.7 onwards should work.. the issue is, it has to be a JDK rather than just a JRE.
We could perhaps check, if it isn't set, whether there's an Android Studio installation and then just use their one
e.g. c:\Program Files\Android\Android Studio\jre:
```
>"c:\Program Files\Android\Android Studio\jre\bin\java" -version
openjdk version "1.8.0_242-release"
OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
OpenJDK 64-Bit Server VM (build 25.242-b01, mixed mode)
```
So, `set JAVA_HOME=c:\Program Files\Android\Android Studio\jre` should work..

**Reply by mehmury** [reply]
Thank you for all ..
i tried compiling again.in last step warning build failed.

i attached adt.log file.

what could i do then?
[adt.log](https://github.com/airsdk/Adobe-Runtime-Support/files/6750556/adt.log)

**Comment by mehmury** [comment]
Finally success compiling!!

I downloaded the latest version adt.jar file published in the forum discussions and its gradle related error was resolved. I tried the aab file I created with bundletool on my android devices. It works fine. thanks.

You can find this adt.jar file in the discussion I gave the link below.

https://github.com/airsdk/Adobe-Runtime-Support/issues/934

Adobe air blend is showing itself as a powerful alternative tool. I am hopeful for the future.

thanks for adobe air harman developer team.!!!..

**Reply by ajwfrost** [reply]
Fixed ADT jar file is now part of the 33.1.1.554 release..
